From 5c0d94d780598a298c9e8685fd811fab9664c368 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Mon, 2 May 2022 12:39:43 -0300
Subject: [PATCH 066/127] linux: Fix posix_spawn return code if clone fails
 (BZ#29109)

The __clone_internal returns the error on errno.

Checked on x86_64-linux-gnu.

(cherry picked from commit 71e2a681f18f617ab962bf8a139bd86d4d440e22)
---
 sysdeps/unix/sysv/linux/spawni.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/spawni.c b/sysdeps/unix/sysv/linux/spawni.c
index d703485e3f..d6f5ca89cd 100644
--- a/sysdeps/unix/sysv/linux/spawni.c
+++ b/sysdeps/unix/sysv/linux/spawni.c
@@ -409,7 +409,7 @@ __spawnix (pid_t * pid, const char *file,
 	__waitpid (new_pid, NULL, 0);
     }
   else
-    ec = -new_pid;
+    ec = errno;
 
   __munmap (stack, stack_size);
 
-- 
2.36.1

